src/Sentry/Protocol/Builders/EnvelopeBuilder.cs

```diff
@@ -30,6 +31,15 @@ public EnvelopeBuilder AddItem(Action<EnvelopeItemBuilder> configure)
             return AddItem(builder.Build());
         }
 
+        public EnvelopeBuilder AddEventItem(SentryEvent @event)
+        {
+            AddHeader("event_id", @event.EventId.ToString());
+            AddItem(i => i
```

Comment:
Ideally we serialize this lazily.
Possibly straight into the stream at the end.
There's no need to be the byte[] filled at this point

Reply:
We'll need it to get `size` (if there's more than 1 item btw) but that can be done when the item gets serialized

Reply:
We can serialize this lazily no problem, but we would have to adapt `SentryEvent` to a similar design that provides `SerializeAsync()` method.

Reply:
Or maybe even better, have `SentryEvent` inherit from `EnvelopeItem`.